packages/slice-machine/tests/pages/cts.spec.tsx

```diff
@@ -430,4 +430,99 @@ describe("Custom Type Builder", () => {
       );
     });
   });
+
+  test("if saving fails a it should not send the save event", async () => {
+    server.use(
+      rest.post("/api/custom-types/save", (_, res, ctx) => {
+        return res(ctx.status(500), ctx.json({}));
+      })
+    );
+    const customTypeId = "a-page";
+
+    singletonRouter.push({
+      pathname: "cts/[ct]",
+      query: { ct: customTypeId },
+    });
+
+    const App = render(<CreateCustomTypeBuilder />, {
+      preloadedState: {
+        environment: {
+          framework: "next",
+          mockConfig: { _cts: { [customTypeId]: {} } },
+        },
+        availableCustomTypes: {
+          [customTypeId]: {
+            local: {
+              id: customTypeId,
+              label: customTypeId,
+              repeatable: true,
+              status: true,
+              tabs: [
+                {
+                  key: "Main",
+                  value: [],
+                },
+              ],
+            },
+          },
+        },
+        selectedCustomType: {
+          model: {
+            id: "a-page",
+            label: "a-page",
+            repeatable: true,
+            status: true,
+            tabs: [
+              {
+                key: "Main",
+                value: [],
+              },
+            ],
+          },
+          initialModel: {
+            id: "a-page",
+            label: "a-page",
+            repeatable: true,
+            status: true,
+            tabs: [
+              {
+                key: "Main",
+                value: [],
+              },
+            ],
+          },
+          mockConfig: {},
+          initialMockConfig: {},
+        },
+      },
+    });
+
+    const addButton = screen.getByTestId("empty-zone-add-new-field");
+    fireEvent.click(addButton);
+
+    const uid = screen.getByText("UID");
+    fireEvent.click(uid);
+
+    const saveFieldButton = screen.getByText("Add");
+
+    await act(async () => {
+      fireEvent.click(saveFieldButton);
+    });
+
+    expect(fakeTracker).toHaveBeenCalledWith(
```

Comment:
This is wrong no ?
Like the tracker shouldn't even had been called if the save to filesystem actually failed.

Reply:
It's part of the set up to make a saveable change to the custom-type